### DISCUSSION TITLE
How do I change the tab favicon?

### DISCUSSION THREAD

**Discussion by Zabooby** [excalith/excalith-start-page#33] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/78969986/235167041-5ad1ee33-0bca-4075-a159-3fdd15864b2f.png)

How would I change the favicon that appears in the tabs?

**Comment by excalith** [comment, marked as answer]
## Answer Update
With v3.0.0, we can use fetch images as favicons, as long as they are either `png`, `svg` or `gif`. For more information, please check out [Customization Wiki Page](https://github.com/excalith/excalith-start-page/wiki/Configuration#Fetch)

## Pre v3.0.0 Answer
Hello @Zabooby, currently you can only change the favicon through cloning this repository. 

In order to do that;
1. [Fork](https://github.com/excalith/excalith-start-page/wiki/Getting-Started#fork) the repository  
2. Head to the `public` folder
3. Replace `favicon.ico` with your own icon
4. Build and deploy / serve etc.
 
Keep in mind that browser caching might prevent updating the icon immediately. You may need to delete caches for your start page manually. 

If this answers your question, please mark it as answered.
Cheers!

**Reply by Zabooby** [reply]
I have forked the repo, removed the `.github/workflows` folder, installed Node.js and Yarn, saved the repo locally, opened and located the folder in the terminal and used the yarn command but am getting this error: 

![image](https://user-images.githubusercontent.com/78969986/235292411-e00d551a-7dd0-41fa-9370-f412f3965d29.png)
PS: I'm very new to this

**Reply by excalith** [reply]
Hey @Zabooby

Seems like you have either downloaded .zip archive or renamed the folder into `excalith-start-page-main`. Just in case, can you please be sure you have the correct file structure within the folder? You can compare it with the original repository contents.

Another thing bugged me is tahat it throws the error for the package name. By default, it is `start-page`. Can you please rename it into `startpage` and give it another try? Haven't encountered such an issue, so not sure about the problem at all.

**Reply by Zabooby** [reply]
Ok I downloaded and extracted from your [releases ](https://github.com/excalith/excalith-start-page/releases/tag/v2.3.1)page just now, I didn't alter any files but I get the same error. 

I also tried renaming to `startpage` but still get the error: 

![image](https://user-images.githubusercontent.com/78969986/235302120-57474a7d-f512-4f0e-b96c-f723cd6156c5.png)

**Reply by excalith** [reply]
I didn't mean you to download from releases. If you want to edit the contents of the project, you should use your forked repository instead. This is the most convinient way for such modifications. By doing that, you will have the absolute control over which updates you want to receive without overriding the code by downloading each new version. Just wanted to let you know about the pipeline since you mentioned that you are new to these stuff.

Back to topic, I was referring changing the `name` of the package in `package.json` file as the error indicates where the issue is, not the directory name. Open the `package.json` file in your favorite code editor, change the `"name": "start-page"` into `"name": "startpage"` and then please try to use `yarn` command. 

if it fails again, can you also share the screenshot of your current directory using `dir` command?

I am pretty sure the problem is not caused by the project since my Windows user friends have successfully deployed their own versions.

**Reply by Zabooby** [reply]
Oh I meant I tried your release to see if my changes where causing the issue. 

I tried again with my [fork ](https://github.com/Zabooby/excalith-start-page) with the name change in `package.json` and still get the error. 

Here is my current directory: 

![image](https://user-images.githubusercontent.com/78969986/235304803-cb2b35cb-0afb-4c2b-ae02-9c3f0750a485.png)

**Comment by Zabooby** [comment]
If I tried using the Docker method would I need Docker running every time I want to use the startpage? 

What are the pros & cons of the Docker method?

**Reply by excalith** [reply]
Hey @Zabooby can you make another thread for Docker? This is a generic question someone might search for later, better to have another topic for it. 

About the issue, I couldn't reproduce this on Windows VM, so I have some wild guesses

- Does your path have special characters? Like your username? 
- Maybe yarn command requires some privilages, equivelant for `sudo` on Linux (Maybe running the terminal as administrator)
- You may need write permissions on that folder, can you try to use a different folder without your OS requiring permissions, maybe Desktop?

Unfortunately thats all I can think of as solutions, you may want to ask `yarn` community for more solutions.

Please mark my first reply regarding the favicon as answer so people can find it faster and let me know if you manage to solve it

Cheers!

**Reply by Zabooby** [reply]
1) My username is just lowercase letters 
2) I've tried running the terminal with admin privileges but still get the error
3) I've tried it in both Desktop and the downloads folder

If I find any answers I'll reply here, thanks for trying to solve this error!

**Reply by excalith** [reply]
Sorry I couldn't help more, will be glad to hear if you find a solution!

**Reply by excalith** [reply]
By the way, just in case you can't find a solution you can always [deploy to web](https://github.com/excalith/excalith-start-page/wiki/Getting-Started#deploying-to-web) using your own fork. You won't need to build it locally, since it will be built on a server such as Vercel automatically when you push to `main` branch.
